test/e2e_node/gpu_device_plugin.go

```diff
@@ -146,19 +150,28 @@ func restartKubelet(f *framework.Framework) {
 	kube := matches[0]
 	framework.Logf("Get running kubelet with systemctl: %v, %v", string(stdout), kube)
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kube).CombinedOutput()
-	if err == nil {
-		return
+	if err != nil {
+		framework.Failf("Failed to restart kubelet with systemctl: %v, %v", err, stdout)
 	}
-	framework.Failf("Failed to restart kubelet with systemctl: %v, %v", err, stdout)
+	Eventually(func() bool {
+		afterSocks, err := filepath.Glob("/var/lib/kubelet/device-plugins/nvidiaGPU*.sock")
+		if err != nil || len(afterSocks) < 1 {
+			return false
+		}
+		framework.Logf("beforeSock %s afterSock %s numberOfGpus %d", beforeSocks[0], afterSocks[0], framework.NumberOfNVIDIAGPUs(getLocalNode(f)))
+		return (afterSocks[0] != beforeSocks[0] && framework.NumberOfNVIDIAGPUs(getLocalNode(f)) > 0)
+	}, 5*time.Minute, time.Second).Should(BeTrue())
```

Comment:
nit: use `framework.Poll` for the poll interval

Reply:
Done.